analyses/sv-analysis/02-shatterseek.R

```diff
@@ -0,0 +1,400 @@
+  colnames(linkss) <- colnames(linkss2)
+  linkss3 <- data.frame()
+  for (id in idlist){
+    ss_id <- linkss2[linkss2$format_id==id,]
+    # caculate the region for the link region
+    linkss$chr <- unique(ss_id$chr)
+    linkss$start <- min(ss_id$start,ss_id$end)
+    linkss$end <- max(ss_id$start,ss_id$end)
+    linkss$chrss_chr <- paste(c(ss_id$chr,ss_id$chrss_chr),collapse="_",sep="_")
+    linkss$sample <- unique(ss_id$sample)
+    linkss$format_id <- unique(ss_id$format_id)
+    linkss$chrss_status <- unique(linkss2[linkss2$format_id==id,]$chrss_status)
+    linkss3 <- rbind(linkss3,linkss)
+  }
+  return(linkss3)
+}
+
+
+# link region analysis
+chrsslink1 <- chrsslink(chrss)
+chrsslink2 <- chrsslink(chrsslink1)
+
+# selfrenew until identical
+round <- 0
+while (!identical(chrsslink1,chrsslink2)){
+  chrsslink1 <- chrsslink(chrsslink2)
+  chrsslink2 <- chrsslink(chrsslink1)
+  chrsslink1$chrss_chr <- sapply(strsplit(chrsslink1$chrss_chr, "_", fixed = TRUE), function(x)
+    paste(unique(x), collapse = "_"))
+  chrsslink2$chrss_chr <- sapply(strsplit(chrsslink2$chrss_chr, "_", fixed = TRUE), function(x)
+    paste(unique(x), collapse = "_"))
+  round <- round+1
+  print(round)
+  print(nrow(chrsslink1)-nrow(chrsslink2))
+}
+
+# split and unique value in chrss_chr
+chrsslink1$chr <- gsub("X","23",chrsslink1$chr)
+chrsslink1$chrss_chr <- gsub("X","23",chrsslink1$chrss_chr)
+chrsslink1$chr <- gsub("Y","24",chrsslink1$chr)
+chrsslink1$chrss_chr <- gsub("Y","24",chrsslink1$chrss_chr)
+chrsslink1$chr <- gsub("M","25",chrsslink1$chr)
+chrsslink1$chrss_chr <- gsub("M","25",chrsslink1$chrss_chr)
+chrsslink1$link_chromosome <- 0
+chrsslink1$chrss_chr <- strsplit(as.character(chrsslink1$chrss_chr), "_", fixed = TRUE)
+for (j in 1:length(chrsslink1$sample)){
+  id <- chrsslink1$sample[j]
+  chr <- chrsslink1$chr[j]
+  chrssid <- chrsslink1[chrsslink1$sample==id,]
+  # sort chrss_chr
+  link_chrss <- as.character(sort(as.numeric(unique(unlist(c(chrssid[chrssid$chr==chr,]$chrss_chr))))))
+  # find intersect rows with link_chrss
+  link_tra <- as.character(sort(as.numeric(unique(unlist(c(chrssid$chrss_chr[unlist(lapply(chrssid$chrss_chr, function(x,y=link_chrss) length(intersect(y, x))>0))]))))))
+  round <- 0
+  while (!identical(link_chrss,link_tra)){
+    link_chrss = link_tra=as.character(sort(as.numeric(unique(unlist(c(chrssid$chrss_chr[unlist(lapply(chrssid$chrss_chr, function(x,y=link_tra) length(intersect(y, x))>0))]))))))
+    link_tra = link_tra=as.character(sort(as.numeric(unique(unlist(c(chrssid$chrss_chr[unlist(lapply(chrssid$chrss_chr, function(x,y=link_chrss) length(intersect(y, x))>0))]))))))
+    round <- round+1
+    print(round)}
+  link_chr <- paste(link_chrss,collapse="_")
+  chrsslink1$link_chromosome[j] <- link_chr
+}
+chrsslink1$chrss_chr <- sapply(chrsslink1$chrss_chr, function(x)
+  paste(unique(x), collapse = "_"))
+chrsslink1$chr <- gsub("23","X",chrsslink1$chr)
+chrsslink1$chr <- gsub("24","Y",chrsslink1$chr)
```

Comment:
oh, actually there only need a step for `X`, because of ShatterSeek only accept `1-22` and `X`, so `Y` and  `M` won't appear in column `chr`.
I will remove `Y`.